src/animation/AnimationUtils.js

```diff
@@ -1,112 +1,171 @@
 		}
 
-		return function equals_primitive( a, b ) {
-			return ( a === b );
-		};
+		return array.slice( from, to );
 
 	},
 
-	clone: function( exemplarValue ) {
+	// converts an array to a specific type
+	convertArray: function( array, type, forceClone ) {
+
+		if ( ! forceClone && array.constructor === type ) return array;
+
+		if ( typeof type.BYTES_PER_ELEMENT === 'number' ) {
+
+			return new type( array );
 
-		var typeName = typeof exemplarValue;
-		if ( typeName === "object" ) {
-			if ( exemplarValue.clone ) {
-				return exemplarValue.clone();
-			}
-			console.error( "can not figure out how to copy exemplarValue", exemplarValue );
 		}
 
-		return exemplarValue;
+		var result = [];
+		result.push.apply( result, array );
+		return result;
 
 	},
 
-	lerp: function( a, b, alpha, interTrack ) {
+	// returns an array by which times and values can be sorted
+	getKeyframeOrder: function( times ) {
 
-		var lerpFunc = THREE.AnimationUtils.getLerpFunc( a, interTrack );
+		function compareTime( i, j ) {
 
-		return lerpFunc( a, b, alpha );
+			return times[ i ] - times[ j ];
 
-	},
+		}
 
-	lerp_object: function( a, b, alpha ) {
-		return a.lerp( b, alpha );
-	},
+		var n = times.length;
+		var result = new Array( n );
+		for ( var i = 0; i !== n; ++ i ) result[ i ] = i;
 
-	slerp_object: function( a, b, alpha ) {
-		return a.slerp( b, alpha );
-	},
+		result.sort( compareTime );
 
-	lerp_number: function( a, b, alpha ) {
-		return a * ( 1 - alpha ) + b * alpha;
-	},
+		return result;
 
-	lerp_boolean: function( a, b, alpha ) {
-		return ( alpha < 0.5 ) ? a : b;
 	},
 
-	lerp_boolean_immediate: function( a, b, alpha ) {
-		return a;
+	// uses the array previously returned by 'getKeyframeOrder' to sort data
+	sortedArray: function( values, stride, order ) {
+
+		var nValues = values.length;
+		var result = new values.constructor( nValues );
+
+		for ( var i = 0, dstOffset = 0; dstOffset !== nValues; ++ i ) {
+
+			var srcOffset = order[ i ] * stride;
+
+			for ( var j = 0; j !== stride; ++ j ) {
+
+				result[ dstOffset ++ ] = values[ srcOffset + j ];
+
+			}
+
+		}
+
+		return result;
+
 	},
 
-	lerp_string: function( a, b, alpha ) {
-		return ( alpha < 0.5 ) ? a : b;
+	// function for parsing AOS keyframe formats
+	flattenJSON: function( jsonKeys, times, values, valuePropertyName ) {
+
+		for ( var i = 0, n = jsonKeys.length; i !== n; ++ i ) {
+
+			var key = jsonKeys[ i ];
+			var value = key[ valuePropertyName ];
+
+			if ( value === undefined ) continue;
+
+			times.push( key[ 'time' ] );
+
+			if ( value[ 'splice' ] !== undefined ) {
+
+				values.push.apply( values, value );
+
+			} else if ( value[ 'toArray' ] !== undefined ) {
+
+				value.toArray( values, values.length );
+
+			} else {
+
+				values.push( value )
+
+			}
+
+		}
+
 	},
 
-	lerp_string_immediate: function( a, b, alpha ) {
- 		return a;
- 	},
+	// fuzz-free, array-based Quaternion SLERP operation
+	slerp: function( dst, dstOffset, src, srcOffset0, srcOffset1, t ) {
```

Comment:
I would put an array-based interpolator functions into their parent class as static functions.  This way when other people want to do fast interpolation they can easily reuse them.  Thus this could be moved to Quaternion.slerpFlat() or somethig like this.